File: 844.cpp
// https://leetcode.com/problems/backspace-string-compare/

class Solution {
public:
    string typed(string s) {
        vector<char> s_list = {};
        for (char c: s) {
            if (c == '#') {
                if (s_list.size() > 0) {
                    s_list.pop_back();
                } 
            }
            else {
                s_list.push_back(c);
            }
        }
        string s_typed(s_list.begin(), s_list.end());
        return s_typed;
    }
    bool backspaceCompare(string s, string t) {
        return typed(s) == typed(t);
    }
};

File: 844.py
# https://leetcode.com/problems/backspace-string-compare/

class Solution:
    def typed(self, s):
        s_list = []

        for char in s:
            if char == "#":
                if len(s_list) > 0:
                    s_list.pop()
            else:
                s_list.append(char)
            
        return "".join(s_list)

    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.typed(s) == self.typed(t)

